llvm-bazel/llvm-project-overlay/clang/BUILD

```diff
@@ -0,0 +1,1976 @@
+        "include/clang/Basic/AttrDocs.td",
+        "include/clang/Basic/DeclNodes.td",
+        "include/clang/Basic/StmtNodes.td",
+    ],
+)
+
+cc_library(
+    name = "serialization",
+    srcs = glob([
+        "lib/Serialization/*.cp" + "p",
+        "lib/Serialization/*.h",
+        "include/clang/Serialization/AttrPCHRead.inc",
+        "include/clang/Serialization/AttrPCHWrite.inc",
+    ]),
+    hdrs = glob([
+        "include/clang/Serialization/*.h",
+    ]),
+    includes = ["include"],
+    textual_hdrs = glob([
+        "include/clang/Serialization/*.def",
+    ]),
+    deps = [
+        ":ast",
+        ":basic",
+        ":lex",
+        ":sema",
+        ":serialization_attr_gen",
+        ":type_nodes_gen",
+        "//llvm:BitReader",
+        "//llvm:BitWriter",
+        "//llvm:BitstreamReader",
+        "//llvm:BitstreamWriter",
+        "//llvm:FrontendOpenMP",
+        "//llvm:Support",
+    ],
+)
+
+cc_library(
+    name = "frontend_tool",
+    srcs = glob([
+        "lib/FrontendTool/*.cpp",
+        "lib/FrontendTool/*.h",
+    ]),
+    hdrs = glob(["include/clang/FrontendTool/*.h"]),
+    includes = ["include"],
+    deps = [
+        ":arc_migrate",
+        ":codegen",
+        ":config",
+        ":driver",
+        ":frontend",
+        ":frontend_rewrite",
+        ":static_analyzer_frontend",
+        "//llvm:Option",
+        "//llvm:Support",
+    ],
+)
+
+cc_library(
+    name = "arc_migrate",
+    srcs = glob([
+        "lib/ARCMigrate/*.cpp",
+        "lib/ARCMigrate/*.h",
+    ]),
+    hdrs = glob(["include/clang/ARCMigrate/*.h"]),
+    includes = ["include"],
+    deps = [
+        ":analysis",
+        ":ast",
+        ":basic",
+        ":edit",
+        ":frontend",
+        ":frontend_rewrite",
+        ":lex",
+        ":parse",
+        ":rewrite",
+        ":sema",
+        ":serialization",
+        ":static_analyzer_checkers",
+        ":static_analyzer_core",
+        "//llvm:Support",
+    ],
+)
+
+cc_library(
+    name = "libclang",
+    srcs = glob([
+        "tools/libclang/*.cpp",
+        "tools/libclang/*.h",
+    ]),
+    hdrs = glob(["include/clang-c/*.h"]),
+    deps = [
+        ":arc_migrate",
+        ":ast",
+        ":basic",
+        ":codegen",
+        ":config",
+        ":driver",
+        ":frontend",
+        ":index",
+        ":lex",
+        ":rewrite",
+        ":sema",
+        ":tooling",
+        "//llvm:BitstreamReader",
+        "//llvm:FrontendOpenMP",
+        "//llvm:Support",
+        "//llvm:config",
+    ],
+)
+
+cc_library(
+    name = "c-bindings",
+    hdrs = glob(["include/clang-c/*.h"]),
+    deps = [
+        ":libclang",
+        #"//clang-tools-extra/clang-include-fixer:include-fixer-plugin",
+    ],
+    alwayslink = 1,
+)
+
+cc_binary(
+    name = "libclang.so",
+    linkopts = [
+        "-nodefaultlibs",
+        "-lc_nonshared",
+        "-lgcc",
+        "-lgcc_eh",
+        "-Wno-unused-command-line-argument",
```

Comment:
These look like they won't work with MSVC

Reply:
True, but will any of this work with MSVC? I would assume we'd need a completely different strategy to build a .dll file here.

Mostly just unsure what the Windows port of this would best look like. Any suggestions on what I should do for now?

Reply:
Yeah I don't know what a real windows port would look like. Just that in other places we've avoided diagnostic flags and such. Probably fine for now. I don't need clang, so my interest in making it build on windows is somewhat lower :grin: